includes/database/queries/class-customer-address.php

```diff
@@ -160,5 +160,35 @@ class Customer_Address extends Query {
 	 */
 	public function __construct( $query = array() ) {
 		parent::__construct( $query );
+		add_action( 'edd_transition_customer_address_type', array( $this, 'demote_primary_addresses' ), 10, 3 );
```

Comment:
Can we move this to `includes/customers/customer-actions.php` instead? (Doesn't yet exist so we'll need to create a new file)

Also, separately, perhaps it would be better to transition on a different column instead. If you edit `includes/database/schemas/class-customer-addresses.php` and find the array for `is_primary`, you can add:

```php
'transition' => true
```

to the array. That will have BerlinDB "listen" on `is_primary`. Then you can hook into `edd_transition_customer_address_is_primary`. Then in the callback function, you can bail if `$new_value` isn't true immediately, without querying the address again.

Reply:
Ahhh, thank you for the `transition` information--that was really puzzling me!

There is already a `customer-actions.php` file  in `includes/admin/customers/customer-actions.php` ... should I just use that instead of creating a new file?

And yes please on the unit tests, @ashleyfae thank you!

Reply:
Never mind on putting the function in the existing file; that only runs in the admin. Noting though that there is a `customer-functions.php` file in just `includes` and maybe it should be moved to `includes/customers`?

Reply:
> Noting though that there is a `customer-functions.php` file in just `includes` and maybe it should be moved to `includes/customers`?

While I'd love to do that for organizational purposes, I think I recall a discussion a while ago about not moving existing files around in case people are referencing them directly. :woman_shrugging: So let's leave it at least for now.

Reply:
Also I'll handle the unit tests once the code is finalized.